api/src/main/java/ai/djl/Device.java

```diff
@@ -143,21 +143,34 @@ public static Device gpu(int deviceId) {
         return of(Type.GPU, deviceId);
     }
 
+    /**
+     * Returns an array of devices.
+     *
+     * <p>If GPUs are available, it will return an array of {@code Device} of size
+     * \(min(numAvailable, maxGpus)\). Else, it will return an array with a single CPU device.
+     *
+     * @return an array of devices
+     */
+    public static Device[] getDevices() {
+        return getDevices(-1);
+    }
+
     /**
      * Returns an array of devices given the maximum number of GPUs to use.
      *
      * <p>If GPUs are available, it will return an array of {@code Device} of size
      * \(min(numAvailable, maxGpus)\). Else, it will return an array with a single CPU device.
      *
-     * @param maxGpus the max number of GPUs to use
+     * @param maxGpus the max number of GPUs to use. Use 0 for no GPUs and -1 for no max.
```

Comment:
I am concerned with what percentage of Java developers know that Integer.MAX_VALUE exists. They can figure it out, but others may just hardcode random big numbers. Whereas -1 treats it as a special case so it may be more straightforward.

Reply:
Per our offline discussion, -1 could be misconstrued to mean either no GPUs or all GPUs if users don't read the docs. Instead, using the newly created `getDevices()` that does not have a maxGpus argument for the all gpu case is clearer than either the -1 or Integer.MAX_VALUE.